### PR TITLE
fixes cocking the gun producing the wrong bullet.

### LINKED ISSUE #3306: Chambered Bullet Bug
## Testmerges:

Server-Side Code Modifications: https://tgstation13.org/parsed-logs/basil/server_side_modifications.dm

Round ID: 1797
BYOND Version: 512.1488
Server revision compiled on: 
Master commit: 2cf92ebfc445c51403818211aed6454e57124001
The following pull requests are currently test merged:
#3300: 'Crash coefficient' by Rohesie at commit c6b87f066d
#3291: 'you are no longer shot by specifically a military laser bolt' by Fikou at commit 405dd03218
#3293: 'Custom Built Shotgun Nerf' by Isy232 at commit 1d9898234c
#3283: 'New spacemandmm release' by spookydonut at commit 18cf9ff355
#3284: 'Minor APC cleanup' by Rohesie at commit 53d2fff134
#3287: 'unfuck shutter element' by spookydonut at commit 5bf010099c
#3280: 'Reverts IS to old storage' by Kuro-ai at commit ca9302db5a
#3282: 'AFK, Respawn and Game tab tweaks' by Rohesie at commit e2cd68497c
#3281: 'Riot Armor Nerf' by Isy232 at commit 605c6f613a
#3274: 'T64 DMR' by David-Stormwell-TGCM at commit 9b480e0d0c
#3261: 'SpecialClick Framework added, Vaulting and holecrawling changed to fit into it' by terranaut1 at commit 39a183a356
#3295: 'Nerfs permastun memes' by Kuro-ai at commit 87d3244755
#3299: 'Lowers runners speed by 0.1.' by Isy232 at commit 5103162f88

Local commit: 9fae0496263fa5b5f4eab1465fc8a824453b2a95
Server tools version: /tg/station 13 Server v3.2.6.2
Don't know if it's from a PR so including this anyways.
## Reproduction:

Right, so this here is a bug related to chambered rounds in magazine using guns. Basically how it happens is, you put a magazine of one type into the empty gun first. Then, you remove the magazine, and insert a different magazine. Now, remove THAT magazine and cock the weapon to remove the chambered round. The chambered round, chambered from the first magazine you loaded in, will now match the SECOND magazine you loaded into the gun, despite the first magazine losing the round from its count. This can be used to create more of certain rare ammo, albeit slowly and laboriously, producing single incendiary shots at a time for things like the M4A3, but also breaks a certain sniper tactic-that of chambering one magazine type and loading another.

### PR DESCRIPTION
**Pull request by Hughgent**
## About The Pull Request

fixes #3306 

## Why It's Good For The Game

slightly more realism, it grabs the bullet in the chamber rather than one from the magazine.

## Changelog
:cl: Hughgent
fix: Cocking a gun now produces the correctly loaded in chamber round.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/updated_projectiles/gun_system.dm [hunk 1]
```diff
@@ -425,12 +425,11 @@ User can be passed as null, (a gun reloading itself for instance), so we need to
 		// Get gun information from the current mag if its equipped otherwise the default ammo & caliber
 		var/bullet_ammo_type
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/gun_system.dm]:
```suggestion
		var/bullet_ammo_type = in_chamber.ammo.type //we know something is in there.
```

File: code/modules/projectiles/updated_projectiles/gun_system.dm [hunk 2]
```diff
@@ -425,12 +425,11 @@ User can be passed as null, (a gun reloading itself for instance), so we need to
...
 		var/bullet_caliber
+		bullet_ammo_type = in_chamber.ammo.type //we know something is in there.
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/gun_system.dm]:
```suggestion
```

File: code/modules/projectiles/updated_projectiles/gun_system.dm [hunk 3]
```diff
@@ -423,14 +423,12 @@ User can be passed as null, (a gun reloading itself for instance), so we need to
 		"<span class='notice'>You cock [src], clearing a [in_chamber.name] from its chamber.</span>", null, 4)
 
 		// Get gun information from the current mag if its equipped otherwise the default ammo & caliber
-		var/bullet_ammo_type
+		var/bullet_ammo_type = in_chamber.ammo.type
 		var/bullet_caliber
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/gun_system.dm]: Alternative syntax, which is nice to know:
```suggestion
		var/bullet_caliber = current_mag ? current_mag.caliber : caliber
```

Comment (Hughgent) [reply code/modules/projectiles/updated_projectiles/gun_system.dm]: I feel the comments are important, but thank you anyways.

**Pull request merged**